sdk/servicebus/azure-messaging-servicebus/src/main/java/com/azure/messaging/servicebus/administration/ServiceBusAdministrationAsyncClient.java

```diff
@@ -2533,6 +2601,27 @@
             });
     }
 
+    /** Adds the additional authentication headers needed for various types of forwarding options.
+     *
+     * @param headerName Name of the auth header
+     */
+    private void addAdditionalAuthHeader(String headerName, HttpHeaders headers) {
+        final String scope;
+
+        if (tokenCredential instanceof ServiceBusSharedKeyCredential) {
+            scope = String.format("https://%s", managementClient.getEndpoint());
+        } else {
+            scope = ServiceBusConstants.AZURE_ACTIVE_DIRECTORY_SCOPE;
+        }
+        final Mono<AccessToken> tokenMono = tokenCredential.getToken(new TokenRequestContext().addScopes(scope));
+        final AccessToken token = tokenMono.block(ServiceBusConstants.OPERATION_TIMEOUT);
```

Comment:
A reactive chain is a flow of transforms that we need to apply to a set of operations to get our final result. This transform necessary in getting the correct headers, so it should be part of a reactor chain (ie. flatMap).  Here is a link for resources: https://projectreactor.io/learn

Reply:
I agree this is not clean. The only way I can think of doing this is by doing an additional HttpPolicy which would open up request body, which would break abstraction.. this is why I went with this approach where we get the token value to be used in the request as part of core buisness logic. Am I missing something here Connie?

Reply:
subscribed to the mono instead of blocking, which is a little better I guess

Reply:
Given this method is called from an "async method," we can return a Mono from this method, which can be chained at the caller side.

Reply:
Thanks Anu and Connie, ended up restructuring the code acording to what I had mentioned of using a standalone supplementary header policy, but using the headers to add the entity details syncronously from the create/update methods (thanks connie for the alternate suggestion)